### PR TITLE
Check

### PR DESCRIPTION
**Pull request by janblazek**
I am adding tests for dnf check command.
Few changes in steps were needed: support for direct installation via rpm command and related changes in repository generation.

### CODE REVIEW

File: dnf-docker-test/features/steps/test_behave.py [hunk 1]
```diff
@@ -93,6 +93,26 @@ def given_repo_condition(context, repo):
         f.write('[' + repo + ']\nname=' + repo + '\nbaseurl=http://127.0.0.1/repo/' + repo + '\nenabled=1\ngpgcheck=0')
 
 
+@step('I execute rpm command "{command}" from repository "{repository}"')
```
Comment (ignatenkobrain) [inline dnf-docker-test/features/steps/test_behave.py]: not a fan of doing special step for rpm commands...

File: dnf-docker-test/features/check-2.feature [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+       Then I execute "dnf" command "check --dependencies" with "fail"
+        And I execute "dnf" command "check --duplicates" with "success"
+        And I execute "dnf" command "check --obsoleted" with "success"
+        And I execute "dnf" command "check --provides" with "success"
+        And I execute "dnf" command "check" with "fail"
+
+  Scenario: Fulfill the missing dependencies
+       When I enable repository "base"
+        And I successfully run "dnf -y install TestB"
+       Then I execute "dnf" command "check --dependencies" with "success"
```
Comment (ignatenkobrain) [inline dnf-docker-test/features/check-2.feature]: usage of deprecated steps is not good idea.

File: dnf-docker-test/features/steps/repo_steps.py [hunk 3]
```diff
@@ -184,12 +184,13 @@ def given_repository_with_packages(ctx, rtype, repository):
         ctx.text = template.render(name=name, **settings)
         fname = "{!s}/{!s}.spec".format(tmpdir, name)
         step_a_file_filepath_with(ctx, fname)
+        buildname = '%{NAME}-%{VERSION}-%{RELEASE}.%{ARCH}.rpm'
```
Comment (ignatenkobrain) [inline dnf-docker-test/features/steps/repo_steps.py]: overriding such stuff from RPM is not good idea.

File: dnf-docker-test/features/check-2.feature (hunk 2 above)
Comment (janblazek) [inline dnf-docker-test/features/check-2.feature]: What's deprecated here?

File: dnf-docker-test/features/steps/test_behave.py (hunk 1 above)
Comment (janblazek) [inline dnf-docker-test/features/steps/test_behave.py]: Is there some possibility for forced installing rpm with broken dependencies via dnf?

File: dnf-docker-test/features/check-2.feature (hunk 2 above)
Comment (ignatenkobrain) [inline dnf-docker-test/features/check-2.feature]: basically, everything what is in test_behave.py is deprecated...

File: dnf-docker-test/features/steps/test_behave.py (hunk 1 above)
Comment (ignatenkobrain) [inline dnf-docker-test/features/steps/test_behave.py]: I don't think so, just saying that probably better is to improve `When I run {command}` by creating another one which will change directory to repo one

**Comment by kkaarreell** [comment]
Let's discuss this feature on Monday as there is some functionality currently missing in the test suite and that should not be directly part of this patch.

**Pull request closed without merging**